### ISSUE TITLE
Add a warning to 3.0's "Getting Started" regarding 2.7's SimpleAnnotationReader in Setup::createAnnotationMetadataConfiguration

### ISSUE THREAD

**Issue by jhard** [doctrine/orm#7965]
TL;DR I believe there should be a warning in 3.0's tutorial that makes it explicit that you shouldn't use it's bootstrap.php if you're using the current 2.7.

Setting up a new project is something I rarely do, so I often just copy paste cli-config.php and bootstrap.php from an existing project and adjust the configuration details.
Since I'm currently working on a console tool, I was trying to set it up "from scratch" to see whether I'd run into any problems with my code. I did!
I jumped into the documentation (on the web) and copy pasted cli-config.php and bootstrap.php, changing only the configuration parameters. This bit me because of SimpleAnnotationReader.

For some reason, I landed in "Getting Started" for 3.0, which uses 
`$config = Setup::createAnnotationMetadataConfiguration(array(__DIR__."/src"), $isDevMode);`
and not 2.7's, which uses 
`$config = Setup::createAnnotationMetadataConfiguration(array(__DIR__."/src"), $isDevMode, $proxyDir, $cache, $useSimpleAnnotationReader);`
with `$useSimpleAnnotationReader` set to false.

There's a small warning up top in 3.0 about this being the documentation for a not-yet-released version, but either I jumped straight to #obtaining-the-entitymanager or banner blindness caused me to not notice this.

Long story short, I ended up with SimpleAnnotationReader being used and not understanding why Doctrine wasn't able to read my entities. This would be less of a problem if the code in 3.0's tutorial actually broke when running in 2.7, because I'd realize something is wrong much more quickly instead of doubting my basic understanding of how to define entities. Instead it "works" but with the unexpected SimpleAnnotationReader. I figured it out in the end but it cost me a few hours and frustration.

It's my fault for not reading the documentation well enough and not see/realize the warning, or not reading the local documentation that is the one relevant to my installed version, but I very much prefer reading in my browser, and I imagine I'm not the only one. I finally found the problem after turning to Google and coming across https://stackoverflow.com/questions/21058689/no-mapped-doctrine-orm-entities-according-to-the-current-configuration 

Since the tutorial is about getting people started, I believe a clear warning in the actual code may be a good idea so they don't start off with problems and frustrations.

When trying to find out how I came to copy from this version in the first place, I also noticed that 3.0's tutorial says "This tutorial assumes you work with Doctrine 2.6 and above. Some of the code will not work with lower versions." which adds to the confusion because the current 2.7 fits that profile but using it with this tutorial gets you started with the SimpleAnnotationReader that will be removed.

I'm not sure what the best approach would be. A comment in the code would be nice, but on the other hand, I rarely read comments when copy pasting code (though I'm sure I would have read it when it didn't work and I started to look for reasons). A check that throws an exception or triggers a warning with an explanation when 3.0's sample bootstrap.php is run in 2.7 would be very nice but there may be scenarios where that leads to problems also. I'm sure you can weigh the options much better than me if you decide my request has merit.

Thank you for reading and Happy Holidays!

**Comment by SenseException** [comment]
It sounds like you're undecided how to handle the mentioning of `SimpleAnnotationReader` yourself. The documentation of 2.7 mentions that the reader shouldn't be used anymore: https://www.doctrine-project.org/projects/doctrine-orm/en/2.7/tutorials/getting-started.html#obtaining-the-entitymanager
Do you have a suggestion how to make this important change more visible? Maybe by repositioning the box where it is mentioned?

BTW, good catch about the 2.x version in the Doctrine 3 docs. This needs to be changed. If you like, you can fix the versions in the Doctrine 3 docs with creating a PR.

**Comment by jhard** [comment]
Even writing about it is confusing, that's how bad it is! ;)
I didn't mean (or wanted!) to use `SimpleAnnotationReader`, I accidentally got an EntityManager that was configured to use it because I copy pasted from the 3.0 docs (which don't include the parameter for `createAnnotationMetadataConfiguration` that suppresses the `SimpleAnnotationReader`, which makes sense, because it doesn't exist in 3.0, hence neither does the parameter, nor does it need a warning about the deprecation), that's what makes this so confusing: using the latest/upcoming version's tutorial with the current stable version will result in "working" code that uses a deprecated feature that will be removed when you're actually using the version the tutorial is for.

It's not a bug, it's just an unlucky combination of circumstances that can be confusing (it was to me at least) because the recommended way to define Entities (which `AnnotationReader` will understand) will not work as expected, but it won't throw errors either, it will simply not find any Entities.

RE changing 2.6 being mentioned: happy to do so, but not sure what to say. It's _technically correct_ that 2.6 is required to run the code, it _does work_ (kind of) with 2.6+, it's just a terrible idea to do run that tutorial's code with < 3.0. Shall we just bump that to say that it assumes to use 3.0? I suppose it makes sense to also replace mentions of "Doctrine 2" with "Doctrine 3", since it's not going to be "Doctrine 2 Version 3.0" and general remarks about the library should use the new name, right?
